backend/src/main/java/com/handong/rebon/shop/domain/tag/application/TagService.java

```diff
@@ -1,7 +1,43 @@
 package com.handong.rebon.shop.domain.tag.application;
 
+import com.handong.rebon.exception.tag.TagExistException;
+import com.handong.rebon.shop.domain.tag.application.dto.TagRequestDto;
+import com.handong.rebon.shop.domain.tag.domain.Tag;
+import com.handong.rebon.shop.domain.tag.domain.repository.TagRepository;
 import org.springframework.stereotype.Service;
+import javax.transaction.Transactional;
+import java.util.List;
 
 @Service
+@Transactional
 public class TagService {
+
+    private final TagRepository tagRepository;
+
+    public TagService(TagRepository tagRepository){
+        this.tagRepository = tagRepository;
+    }
+
+    public Long createTag(TagRequestDto tagRequestDto) {
+        String tagName = tagRequestDto.getName();
+        validateDuplicateTag(tagName);
+
+        Tag newTag = Tag.builder()
+                .name(tagName)
+                .build();
+
+        Tag savedTag = tagRepository.save(newTag);
+        return savedTag.getId();
+    }
+
+    private void validateDuplicateTag(String name) {
+        if (tagRepository.findByName(name).isPresent()){
+            throw new TagExistException();
+        }
+    }
```

Comment:
findByName을 통해 검증을 하고 싶다면 이렇게 줄일 수 있겠네요
```suggestion
    private void validateDuplicateTag(String name) {
        tagRepository.findByName(name)
                                .orElseThrow(TagExistException::new);
    }
```
```suggestion
    private void validateDuplicateTag(String name) {
        if (tagRepository.existsByName(name)) {
            throw new TagExistException();
        }
    }
```

Reply:
아 이렇게 더 코드가 간결해질 수 있겠군요!! 감사합니다 :)